src/main/java/com/google/devtools/build/lib/skyframe/ActionMetadataHandler.java

```diff
@@ -483,6 +485,18 @@ public void injectRemoteFile(Artifact output, byte[] digest, long size, int loca
     store.injectRemoteFile(output, digest, size, locationIndex);
   }
 
+  @Override
+  public void injectRemoteDirectory(SpecialArtifact output,
+      Map<PathFragment, RemoteFileArtifactValue> children) {
+    ImmutableMap.Builder<TreeFileArtifact, FileArtifactValue> childFileValues = ImmutableMap.builder();
+    for (Map.Entry<PathFragment, RemoteFileArtifactValue> child : children.entrySet()) {
```

Comment:
Check here that they are actually children of the tree artifact.

Reply:
They must always be children because the PathFragment is interpreted as relative to the parent. So no matter what value PathFragment has it will be interpreted as relative to the path of `treeArtifact`.